src/catch2/catch_config.hpp

```diff
@@ -20,6 +22,25 @@ namespace Catch {
     struct IStream;
 
     struct ConfigData {
+        struct ReporterAndFile {
+            std::string reporterName;
+
+            // If none, the output goes to the default output.
+            Option<std::string> outputFileName;
+
+            friend bool operator==(ReporterAndFile const& a, ReporterAndFile const& b) {
+                return a.reporterName == b.reporterName && a.outputFileName == b.outputFileName;
+            }
+
+            friend std::ostream& operator<<(std::ostream &os, ReporterAndFile const& reporter) {
```

Comment:
I would rather demote the definition (and the related include) into the .cpp file.

Reply:
Also, is this used anywhere that is not the self tests? If not, I would prefer a `StringMaker` specialization there.

Reply:
Demoted. I'll check where it's needed.